src/db/spanner/models.rs

```diff
@@ -869,13 +996,17 @@ impl SpannerDb {
             .execute_dml_async(&self.conn)
             .await?;
         if affected_rows > 0 {
+            let mut tags = Tags::default();
+            tags.tags.insert("collection".to_string(), collection_id);
+            self.metrics
+                .incr_with_tags("storage.spanner.delete_collection", Some(tags));
             self.erect_tombstone(&params.user_id).await
         } else {
             self.get_storage_timestamp(params.user_id).await
         }
     }
 
-    pub(super) async fn touch_collection_async(
+    pub(super) async fn update_collection_async(
```

Comment:
The writes always call touch_collection first to ensure a parent record exists in user_collections (because of interleaving). Switching touch_collection to calculating the quota means we're calculating it before writing new data.

Reply:
I think your recent reply on the bug is in reference to this?

Reply:
Ah, thanks! I missed moving that metric call to `calc_quota_usage_async` which does the quota measurement. `update_collection_async` in spanner still just deals with modified only.